shared/constants/platform.desktop.js

```diff
@@ -56,10 +56,12 @@ function win32SocketDialPath (): string {
 }
 
 function linuxSocketDialPath (): string {
-  if (runMode === 'prod') {
-    return path.join(`${getenv('XDG_RUNTIME_DIR', '')}/keybase/`, socketName)
-  }
-  return path.join(`${getenv('XDG_RUNTIME_DIR', '')}/keybase.${runMode}/`, socketName)
+  // If XDG_RUNTIME_DIR is defined use that, else use $HOME/.config.
+  const homeCacheDir = path.join(getenv('HOME', ''), '.config')
```

Comment:
The empty string fallback here would cause confusing errors, right? (Especially confusing because I think they'll only repro if your terminal isn't currently cd'd into your home folder.) Should we make this return an error one way or another if `HOME` is not defined?